src/java.base/share/classes/java/net/InetSocketAddress.java

```diff
@@ -409,7 +409,7 @@ public String toString() {
      */
     @Override
     public final boolean equals(Object obj) {
-        if (obj == null || !(obj instanceof InetSocketAddress))
+        if (!(obj instanceof InetSocketAddress))
             return false;
         return holder.equals(((InetSocketAddress) obj).holder);
```

Comment:
If we restructure this a little we can get:

    public final boolean equals(Object obj) {
        if (obj instanceof InetSocketAddress that)
            return holder.equals(that.holder);
        return false;
    }